aws/resource_aws_ec2_client_vpn_network_association_test.go

```diff
@@ -52,6 +52,26 @@ func TestAccAwsEc2ClientVpnNetworkAssociation_disappears(t *testing.T) {
 	})
 }
 
+func TestAccAwsEc2ClientVpnNetworkAssociation_securityGroups(t *testing.T) {
+	var assoc1 ec2.TargetNetwork
+	rStr := acctest.RandString(5)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProvidersWithTLS,
+		CheckDestroy: testAccCheckAwsEc2ClientVpnNetworkAssociationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEc2ClientVpnNetworkAssociationSecurityGroups(rStr),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsEc2ClientVpnNetworkAssociationExists("aws_ec2_client_vpn_network_association.test", &assoc1),
+					resource.TestCheckResourceAttr("aws_ec2_client_vpn_network_association.test", "security_groups.#", "2"),
+				),
+			},
```

Comment:
We should include another `TestStep` here that performs some form of update 👍

Reply:
One thing I'm noticing with the Update function is that `ApplySecurityGroupsToClientVpnTargetNetwork` call doesn't zero out `security_groups` it just replaces ones that are already there.  Going from 2 security groups to zero means that the call ignores the empty request and leaves the security groups untouched.  Weird, but not unmanageable.

Reply:
Sorry that was what I was trying to say in https://github.com/terraform-providers/terraform-provider-aws/pull/7500#discussion_r255805756

Reply:
Ok, I think I see what you are saying.  See my latest commit.  I was able to get the behavior I'm looking for by making `security_groups` a required parameter (to account for the API adding the default VPC security group in the absence of custom groups).  Not ideal but it works.

Reply:
Scratch that, Travis doesn't like the `Required` + `Computed` part.